\section{Introduction} 
\label{sec:introduction} 

In recent decades there was a significant growth of biological data generated by experimental techniques such as the new generation DNA sequencing, protein sequencing and protein structure determination. Much of these data is organized and made publicly available to the scientific community in biological databases over the Internet. According to \cite{lesk2005database} these repositories not only store biological raw data but also relevant information such as literature data, protein function, relationship between a protein and its encoding gene, among other metadata.

Given that these biological databases are growing at very high rates, most of these metadata are automatically assigned. In most cases, with no laboratory experiments at all, the roles of most genes in several organisms have been reported by homology propagation \cite{brenner1999errors}. To ensure that these annotations remain reliable, studies about the confiability of the entries, together with measures of confidence should be developed. Many studies have drawn attention to errors rates in the biological databases annotations \cite{devos2001intrinsic,green2005genome,gilks2005percolation,jones2007estimating,schnoes2009annotation,hung2010detect}.

In fact, the automatic identification of theses errors is still an open problem and several challenges must be overcome. Without laboratory experiments to automatically verify assigned annotations it will remain impossible to establish a definite conclusion. However, most of the studies present comparisons of a diversity of methods of functional annotation, and show that they are widely incompatible, constraining their accuracy. 

A major step toward automatic error detection is a description of how and to what extent biological databases entries annotations evolve. In other words, we must fully understand why some entries seem to be more stable while others remain more volatile, and also what are the factors that determine these constrating behaviors. %[SABRINA] behaviors

The research and development of models and algorithms, coupled with constantly improving visualization resources, represent a promising approach toward understanding how biological databases evolve. Interactive visualizations can be specially powerful in depicting in a macro/micro perspective voluminous, high-dimensional and complex datasets and also in helping users unveil trends and exceptions in those datasets. 

\subsection{Enzyme annotations}

By the late 1950's it had become evident that the nomenclature of enzymology, in a period when the number of known enzymes was increasing rapidly, was getting out of hand. In many cases the same enzymes became known by several different names, while conversely the same name was sometimes given to different enzymes. Many of the names conveyed little or no idea of the nature of the reactions catalysed, and similar names were sometimes given to enzymes of quite different types. To meet this situation, the General Assembly of the International Union of Biochemistry (IUB) decided, in consultation with the International Union of Pure and Applied Chemistry (IUPAC), to set up an International Commission on Enzymes. Its objective was to consider the classification and nomenclature of enzymes and co-enzymes, their units of activity and standard methods of assay, together with the symbols used in the description of enzyme kinetics. The Commission prepared a report in 1961 that was promptly adopted and, since then, has been widely used in scientific journals, textbooks, etc. The size of the Enzyme Commission number (EC number) list has increased steadily since the publication of the first report and also many corrections were done.

The EC number is a numerical classification scheme for enzymes, based on the chemical reactions they catalyze. Every enzyme code consists of four numbers separated by periods. Those numbers represent a hierarchical progressively finer classification of the catalized reaction. For example, the code: 3.4.21.4 is a:
\begin{description}
\item [3:] hydrolase, which means the enzyme breaks a chemical bond using a water molecule.
\item [3.4:] peptidase, which means the broken bond is a peptide bond, i.e., a bond between amino acids in a protein chain.
\item [3.4.21:] endopeptidase, because it breaks an intra-chain peptide bond.
\item [3.4.21.4:] trypsin, because enzyme has the specificity of cutting close the residues arginine and lisine.
\end{description}

%When a new enzyme is annotated, one can add from one to four levels of the EC number, depending on the detail of existing knowledge. In the better case, we know all about the catalyzed reaction as well as the specific substrates and products involved. However, in many cases all we know is that the molecule is an enzyme. In this case, the annotation is left "-.-.-.-". An EC number "3.4.21.-", for instance, means we don't know enzyme substrates specifically although we have information about the reaction catalyzed.

%[SABRINA] tirei -.-.-.- porque isso não existe na base, nós é que modelamos dessa forma para representar entradas sem ec.

When a new enzyme is annotated, one can add from one to four levels of the EC number, depending on the detail of the existing knowledge. In the best scenario, everything is knwon about the catalyzed reaction algon with the specific substrates and products involved. However, in many cases, when not all the details about the catalytic activity are known, partial EC numbers, in which hyphens are written in the unknown levels, are used to annotate enzymes. The EC number "3.4.21.-", for instance, means that the specific enzyme substrates are not knwon, although information about the reaction catalyzed is available.
\\
\\
In this paper, we tackle the problem of analyzing enzyme annotation dynamics and propose a technique to visualize the evolution of these annotations across several releases of UniProt/SwissProt databases. This paper is organized as follows: in section \ref{sec:modelling}, we describe how we modeled the problem. Section \ref{sec:dataset} details the dataset presented in the visualization. In section \ref{sec:related_work}, we talk about previous related researches and in section \ref{sec:technique}, we describe in detail the basis of the technique proposed as well as its capabilities. Finally, we discuss several insights we obtained in section \ref{sec:discussion} and conclude the work.
